Pinta.Core/Classes/AsyncEffectRenderer.cs

```diff
@@ -170,73 +170,92 @@ internal void Dispose ()
 		timer_tick_id = 0;
 	}
 
+	CancellationToken ReplaceCancellationSource ()
+	{
+		CancellationTokenSource newSource = new ();
+		CancellationTokenSource oldSource = Interlocked.Exchange (ref cancellation_source, newSource);
```

Comment:
Is the `Interlocked.Exchange` necessary? I'm fairly sure `cancellation_source` will only be accessed by the main thread

Reply:
@cameronwhite if that's the case then I can see why `Interlocked.Exchange` is not necessary.

I've removed the references to it in the last commit.

I think, though, that even if it's not needed, it could make the intent clearer, as everything happens in a single statement instead of two and we also express that we want it to be thread-safe (even if it already is).

If you agree, you can discard the last commit, and if you don't, you can keep the commit.

Reply:
Thanks, I think I prefer removing the Interlocked.Exchange just to make it clear that we don't expect this to be called concurrently.